crates/extensions/tedge_config_manager/src/actor.rs

```diff
@@ -358,22 +392,38 @@ impl ConfigManagerWorker {
         let from_path = Utf8Path::from_path(&from)
             .with_context(|| format!("path is not utf-8: '{}'", from.to_string_lossy()))?;
 
-        let file_entry = self
-            .plugin_config
-            .get_file_entry_from_type(&request.config_type)?;
-        let to = Utf8PathBuf::from(&file_entry.path);
-
-        if let Some(parent) = to.parent() {
-            if !parent.exists() {
-                self.create_parent_dirs(parent, file_entry)?;
-            }
-        }
+        let config_path =
+            if let Some((config_type, plugin_name)) = request.config_type.split_once("::") {
+                if let Some(plugin) = self.external_plugins.by_plugin_type(plugin_name) {
+                    plugin.set(config_type, from_path.as_std_path()).await?;
+                    plugin.finalize(config_type).await?;
+                    None
```

Comment:
After writing this (`set` and `finalize` invoked back-to-back), I'm having second thoughts about this `finalize` command. Why do we even need a separate command like that. Why not perform these steps as part of the `set` command itself. I don't really see any clear benefits in having it separately, while both steps are performed as part of the same `executing` stage itself.

Reply:
I'm okay to remove the `finalize` action and to mandate the `set` action to make the changes effective.

It will be a non-breaking change to add a `finalize` action if seen after-all useful.